working/augmentation-libraries/feature-specification.md

```diff
@@ -122,30 +122,34 @@ syntax][import]?**
 
 [import]: https://github.com/dart-lang/language/blob/master/working/0649%20-%20Import%20shorthand/proposal.md
 
-The URL points to the main library that this augmentation is applied to.
+The URL points to the augmented library that this augmentation is applied to,
+which must be a regular library (and not a library augmentation).
 
 After that, an augmentation may contain anything a regular Dart library can
 contain: imports, exports, classes, functions, constants, etc. All
-augmentations of a main library share the same top level declaration scope.
-Declarations in any augmentation or the main library are visible to all of the
-others, including private ones.
+augmentations of an augmented library share the same top level declaration
+scope. Declarations in any library augmentation or the augmented library are
+visible to all of the others, including private ones.
 
-However, augmentations do *not* share an import scope with the main library or
-each other. The libraries one augmentation imports are visible only to that
-file.
+However, library augmentations do *not* share an import scope with the augmented
+library or each other. The libraries one library augmentation imports are
+visible only to that file.
 
 It is a compile-time error if:
 
-*   A top-level declaration in an augmentation has the same name as a
-    declaration in the main library or another of its augmentations (unless it
-    is an *augmenting* declaration, described below). *This is the same error
-    conceptually as having a name collision in one file.*
+*   A top-level declaration in a library augmentation has the same name as a
+    declaration in the augmented library or another of its library augmentations
+    (unless it is a declaration augmentation, described below). *This is the
+    same error conceptually as having a name collision in one file.*
 
 *   An augmentation contains any `part` directives.
 
+*   A library augmentation contains a normal `library` directive. They are not
+    self-contained libraries, only pieces of the augmented library.
+
 ### Applying an augmentation
 
-A main library applies an augmentation to itself using a new `augment` directive
+A library applies an augmentation to itself using a new `augment` directive
 which looks like this:
 
 ```dart
```

Comment:
I'm iffy on this syntax. It reads to me like it's saying that *this* file is augmenting `'some_augmentation.dart'` when it's the exact opposite.

This used to be `import augment`, right? I think I still lean towards that syntax.

Reply:
Yeah, I will probably revert this part, as I agree it is a bit weird. I don't like either of them really, but I don't have an alternative proposal either 🤷‍♂️ .